snapshot.sh

```diff
@@ -16,90 +32,84 @@ function help_menu() {
 "
 }
 
-NETWORK=
-S3_TENANT_ID=
-S3_ENDPOINT=
-S3_BUCKET=
-DAEMON=
-USER_DIR=
-
-# getopt boilerplate for argument parsing
-OPTS=$(getopt -o b:e:i:n:d:u:h --long bucket:,endpoint:,id:,network:,daemon:,userdir:,help \
+make_opts() {
+    # getopt boilerplate for argument parsing
+    OPTS=$(getopt -o b:e:i:n:d:u:h --long bucket:,endpoint:,id:,network:,daemon:,userdir:,help \
             -n 'Crypto Chemistry Snapshot Uploader' -- "$@")
+    [[ $? != 0 ]] && { echo "Terminating..." >&2; exit 51; }
+    eval set -- "${OPTS}"
+}
+
+parse_args() {
+    while true; do
+    case "$1" in
+        -b | --bucket ) S3_BUCKET="$2"; shift 2 ;;
+        -e | --endpoint ) S3_ENDPOINT="$2"; shift 2 ;;
+        -i | --id ) S3_TENANT_ID="$2"; shift 2 ;;
+        -n | --network ) NETWORK="$2"; shift 2 ;;
+        -d | --daemon ) DAEMON="$2"; shift 2 ;;
+        -u | --userdir ) USER_DIR="$2"; shift 2 ;;
+        -h | --help ) HELP_MENU="True"; shift ;;
+        -- ) shift; break ;;
+        * ) break ;;
+    esac
+    done
+
+    [[ ! -z $HELP_MENU ]] && { help_menu; exit 0; }
+
+    if [[ -z $S3_BUCKET || -z $S3_ENDPOINT || -z $S3_TENANT_ID || -z $NETWORK || -z $DAEMON || -z $USER_DIR ]]; then
+        printf "\
+        ${SCRIPT_NAME}: Error - Missing Arguments
+        The following arguments are required:
+            -b, --bucket
+            -e, --endpoint
+            -i, --id
+            -n, --network
+            -d, --daemon
+    "
+        exit 52
+    fi
+
+    # Ensure endpoint URL does not have a single trailing '/'
+    S3_ENDPOINT="${S3_ENDPOINT%/}"
+}
+
+get_block_height() {
+    # Service must be running to get block height:
+    systemctl start cosmovisor.service >/dev/null && \
+    BLOCK_HEIGHT=$(curl -s http://localhost:26657/status | jq -r .result.sync_info.latest_block_height)
+    # CC: Is it necessary to stop the service at this time?
```

Comment:
It is necessary to stop the service otherwise you can end up with some corrupted snapshots. It won't always be an issue, but it happens enough from my experience to be worth stopping the service.

Reply:
cool